### ISSUE TITLE
Add option to ignore ::before and ::after to font-family-no-missing-generic-family-keyword

### ISSUE THREAD

**Issue by EvgenyOrekhov** [stylelint/stylelint#3048]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A feature request to add a new option to ignore `::before` and `::after` to the `font-family-no-missing-generic-family-keyword` rule.

I declare icons in my CSS using `::before` and `::after`.
I don't think it makes sense to use a generic family keyword in those cases.
I'd like for `font-family-no-missing-generic-family-keyword` to be able to ignore those cases.

> Which rule, if any, is this issue related to?

`font-family-no-missing-generic-family-keyword`

> What CSS is needed to reproduce this issue?

```css
.users::before {
  font-family: "FontAwesome";
  content: "\f0c0";
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "font-family-no-missing-generic-family-keyword": true
  }
}
```

> Which version of stylelint are you using?

`8.3.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint *.css`.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
2:18    Unexpected missing generic font family (font-family-no-missing-generic-family-keyword)
```

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

I doubt the necessity of this option for a majority of our users. It's quite specific use case. I agree with what @jeddy3 [said](https://github.com/stylelint/stylelint/issues/3033#issuecomment-347006184) in a similar issue:

> > xPaw: Side question, should I still be adding a generic font for icon fonts?
>
> I'm not 100% sure, I'm afraid. I doubt [the spec](https://www.w3.org/TR/css-fonts-3/#generic-font-families) anticipated them at the time of writing. I guess there's no harm in adding a generic font after an icon font. It also feels like icon fonts were a workaround for the poor support of SVGs in some browsers, which I believe is no longer the case. So, icon fonts might become an artefact of their time.

**Comment by jeddy3** [comment]
Yes, I think that comment is applicable here also. There's no harm adding a generic font after an icon font (or use `/* stylelint-disable */` comments). And I suspect most users are migrating away from icon fonts. I'll close this issue for now, but we can always reopen it if more users request a similar option.
